### PR TITLE
Add notifications

### PR DESCRIPTION
**Pull request by geshuming**
Notifications. Finally.

Core features:
- Opened 2 new endpoints. `Fetch` and `Acknowledge`
    - `Fetch` allows users to fetch all outstanding (unread) notifications
    - `Acknowledge` allows users to acknowledge a specific notification, setting it to read.
- Types of notifications
    - `new` - represents assessments that are new to students
    - `autograded` - represents assessments that have been autograded
    - `graded` - represents assessments that have been manually graded
    - `unsubmitted` - represents assessments that have been unsubmitted by the grading staff
    - `submitted` - represents submissions that have been submitted by the student

### CODE REVIEW

**Comment by jiachen247** [comment]
Can consider how we can integrate with SES or SNS to tele or smth for realtime feedback so can close the feedback loop even more!

File: lib/cadet/accounts/notifications.ex [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+          read: false,
+          role: :student
+        })
+    end
+    |> Repo.insert_or_update()
+  end
+
+  defp write_student(params), do: {:error, Notification.changeset(%Notification{}, params)}
+
+  defp write_avenger(
```
[the remark this answers is not in the document]
Comment (geshuming) [reply lib/cadet/accounts/notifications.ex]: Done

**Comment by geshuming** [comment]
Integrated with chatkit

File: lib/cadet/accounts/notification_type.ex [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+defenum(Cadet.Accounts.NotificationType, :notification_type, [
+  # Notifications for new assessments
+  :new,
+
+  # Notifications for autograded assessments
+  :autograded,
+
+  # Notifications for manually graded assessments
+  :graded,
```
Comment (jiachen247) [inline lib/cadet/accounts/notification_type.ex]: can standardize the spacing here!

**Pull request merged**